zkevm-circuits/src/lib.rs

```diff
@@ -5,6 +5,8 @@
 #![allow(incomplete_features)]
 // Needed by DummyGadget in evm circuit
 #![feature(adt_const_params)]
+#![allow(clippy::type_complexity)]
```

Comment:
Can this be added to the affected types only, instead of enabling it globally so that it's not abused?

Reply:
Makes sense! Moved it to the struct and the `block_modifiers` fns